front-end/src/services/subscriptionService.js

```diff
@@ -0,0 +1,32 @@
+import API, { API_URL } from 'API';
+
+export const getAllUserSubscriptions = async (id) => {
+  const { data } = await API.get(`${API_URL}/api/subscriptions/user/${id}`);
+  if (id === undefined) {
+    return [];
+  }
+  return data;
+};
+
+export const createSubscription = async (subscription) => {
+  const body = {
+    teacher: {
+      _id: subscription.teacher._id,
```

Comment:
Why we need _id, mongodb creating id automatically

Reply:
It is teacher id